src/dnx.coreclr/dnx.coreclr.cpp

```diff
@@ -35,344 +27,187 @@ bool GetTrustedPlatformAssembliesList(const wchar_t* szDirectory, bool bNative,
-    DWORD dwCoreClrDirectory = GetEnvironmentVariableW(L"CORECLR_DIR", szCoreClrDirectory, MAX_PATH);
-    HMODULE hCoreCLRModule = nullptr;
-
-    if (dwCoreClrDirectory != 0)
+    auto loader_module = LoadLoaderModule(trace_writer);
+    if (!loader_module)
     {
-        WCHAR wszClrPath[MAX_PATH];
-        wszClrPath[0] = L'\0';
-
-        errno = wcscpy_s(wszClrPath, _countof(wszClrPath), szCoreClrDirectory);
-        CHECK_RETURN_VALUE_FAIL_EXIT_VIA_FINISHED(errno);
-
-        if (wszClrPath[wcslen(wszClrPath) - 1] != L'\\')
-        {
-            errno = wcscat_s(wszClrPath, _countof(wszClrPath), L"\\");
-            CHECK_RETURN_VALUE_FAIL_EXIT_VIA_FINISHED(errno);
-        }
-
-        errno = wcscat_s(wszClrPath, _countof(wszClrPath), L"coreclr.dll");
-        CHECK_RETURN_VALUE_FAIL_EXIT_VIA_FINISHED(errno);
-
-        //Scan through module name list looking for a valid module that has first DLL export
-        rgwszModuleFileName = rgwzOSLoaderModuleNames[dwModuleFileName];
-        while (rgwszModuleFileName != NULL)
-        {
-            fSuccess = KlrLoadLibraryExWAndGetProcAddress(
-                            rgwszModuleFileName,
-                            pszAddDllDirectoryName,
-                            &hOSLoaderModule,
-                            (FARPROC*)&pFnAddDllDirectory);
-            if (fSuccess)
-                break;
-
-            dwModuleFileName++;
-            rgwszModuleFileName = rgwzOSLoaderModuleNames[dwModuleFileName];
-         }
-
-        if (!hOSLoaderModule || !pFnAddDllDirectory)
-        {
-            fSuccess = false;
-            goto Finished;
-        }
-
-        //Find the second DLL export
-        pFnSetDefaultDllDirectories = (FnSetDefaultDllDirectories)::GetProcAddress(hOSLoaderModule, pszSetDefaultDllDirectoriesName);
-        if (!pFnSetDefaultDllDirectories)
-        {
-            if (m_fVerboseTrace)
-                ::wprintf_s(L"Failed to find function %S in %s\n", pszSetDefaultDllDirectoriesName, rgwszModuleFileName);
-
-            fSuccess = false;
-            goto Finished;
-        }
-
-        //Verify HANDLE and two DLL exports are valid before proceeding
-        if (!hOSLoaderModule || !pFnAddDllDirectory || !pFnSetDefaultDllDirectories)
-        {
-            fSuccess = false;
-            goto Finished;
-        }
-
-        pFnAddDllDirectory(szCoreClrDirectory);
-        // Modify the default dll flags so that dependencies can be found in this path
-        pFnSetDefaultDllDirectories(LOAD_LIBRARY_SEARCH_DEFAULT_DIRS | LOAD_LIBRARY_SEARCH_USER_DIRS);
-
-        fSuccess = true;
-
-        // Continue loading as usual
-        hCoreCLRModule = ::LoadLibraryExW(wszClrPath, NULL, 0);
+        trace_writer.write(L"Failed to load loader module", false);
+        return nullptr;
     }
 
-    if (hCoreCLRModule == nullptr)
+    auto AddDllDirectory_function = (FnAddDllDirectory)GetProcAddress(loader_module, "AddDllDirectory");
```

Comment:
I don't want to admit it but it is and the replacement I came up with is not even close. I will change it back to `pfn...`